.github/workflows/run_tests.yml

```diff
@@ -0,0 +1,37 @@
+name: Automated Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - '*'
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
```

Comment:
This leads to conflicts because the project is only meant to run on 3.11 and higher. That means the pyproject.toml has to be changed to run this correctly.